pkg/controllers/controllers.go

```diff
@@ -52,7 +54,9 @@ func NewControllers(ctx context.Context, sess *session.Session, clk clock.Clock,
 	pricingProvider *pricing.Provider, amiProvider *amifamily.Provider, launchTemplateProvider *launchtemplate.Provider) []controller.Controller {
 
 	controllers := []controller.Controller{
-		nodeclass.NewController(kubeClient, recorder, subnetProvider, securityGroupProvider, amiProvider, instanceProfileProvider, launchTemplateProvider),
+		hash.NewController(kubeClient),
```

Comment:
```suggestion
		nodeclasshash.NewController(kubeClient),
```

Like the other controllers, I think that it's a bit easier to read if we prefix these with the `nodeclass` package name as well